presto-raptor/src/main/java/com/facebook/presto/raptor/storage/StorageService.java

```diff
@@ -23,13 +24,13 @@
 
     long getAvailableBytes();
 
-    void createParents(File file);
+    void createParents(Path file);
 
-    File getStorageFile(UUID shardUuid);
+    Path getStorageFile(UUID shardUuid);
 
-    File getStagingFile(UUID shardUuid);
+    Path getStagingFile(UUID shardUuid);
 
-    File getQuarantineFile(UUID shardUuid);
+    Path getQuarantineFile(UUID shardUuid);
```

Comment:
Could this all be framed instead in terms of URI, and we switch off the URI scheme to determine the FileSystem to use?

Reply:
I though about that. The reason to have `Path` is to work with `FileSystem` class (which all other Presto file systems like S3, GCS inherit from). That helps with `FileSystem` interfaces including `create`, `open`, `exists`, `delete`, `rename`, etc without converting from/to URI. Also, `Path` can support `getFileSystem` from a config as well. After checking how `HdfsEnvironment` class is used in our codebase, I decided to use `Path`/`FileSystem` combination. What do you think?

Reply:
Internally, Path contains the URI and will internally switch off the scheme.  So my suggestions was not that useful.